### PR TITLE
fix: unexpected eth_getBalance calls

### PR DESCRIPTION
**Pull request by 0xmkh**
releated to this: https://github.com/rainbow-me/rainbowkit/issues/1488

### CODE REVIEW

File: packages/rainbowkit/src/components/ConnectButton/ConnectButton.tsx [hunk 1]
```diff
@@ -43,8 +43,13 @@ export function ConnectButton({
...
   const i18n = useContext(I18nContext);
 
+  const shouldShowBalance =
+    normalizeResponsiveValue(showBalance)[
+      isMobile() ? 'smallScreen' : 'largeScreen'
+    ];
+
   return (
-    <ConnectButtonRenderer>
+    <ConnectButtonRenderer showBalance={shouldShowBalance}>
```
Comment (KosmosKey) [inline packages/rainbowkit/src/components/ConnectButton/ConnectButton.tsx:52]: I added an extra prop to the `ConnectButtonRendered` component called `showBalance`. We can pass either `true` or `false` and it will make sure to fetch the balance only if it's `true`.

File: packages/rainbowkit/src/components/AccountModal/AccountModal.tsx [hunk 2]
```diff
@@ -13,10 +13,11 @@ export interface AccountModalProps {
 
 export function AccountModal({ onClose, open }: AccountModalProps) {
   const { address } = useAccount();
-  const { data: balanceData } = useBalance({ address });
   const ensName = useMainnetEnsName(address);
   const ensAvatar = useMainnetEnsAvatar(ensName);
   const { disconnect } = useDisconnect();
+  const [balanceData, setBalanceData] =
+    useState<ReturnType<typeof useBalance>['data']>();
```
Comment (KosmosKey) [inline packages/rainbowkit/src/components/AccountModal/AccountModal.tsx:20]: Not sure if it makes sense to do another PR for this, but the chain modal would also call `"eth_getBalance"` randomly. I made sure to only call `"eth_getBalance"` if the chain modal is open.

File: packages/rainbowkit/src/components/ProfileDetails/ProfileDetails.tsx [hunk 3]
```diff
@@ -32,10 +33,21 @@ export function ProfileDetails({
...
+  const { data: wagmiBalanceData } = useBalance({
+    address,
+    enabled: !balanceData,
+  });
+
+  // biome-ignore lint/nursery/useExhaustiveDependencies: <explanation>
+  useEffect(() => {
+    if (wagmiBalanceData) setBalanceData(wagmiBalanceData);
+  }, [wagmiBalanceData?.value]);
+
```
Comment (DanielSinclair) [inline packages/rainbowkit/src/components/ProfileDetails/ProfileDetails.tsx:50]: Does the `balanceData` prop that's passed down here trigger the same behavior upon re-render?

File: packages/rainbowkit/src/components/ConnectButton/ConnectButton.tsx (hunk 1 above)
Comment (DanielSinclair) [reply packages/rainbowkit/src/components/ConnectButton/ConnectButton.tsx:52]: I think we'll want to avoid this addition since it exposes a new prop to the `ConnectButton.Custom` API

Comment (DanielSinclair) [reply packages/rainbowkit/src/components/ConnectButton/ConnectButton.tsx:52]: Not sure the best way to handle this. I wonder if we should create something similar to `ShowRecentTransactionsContext` that's exposed in `RainbowKitProvider` to requested limited RPC calls. That seems like the root cause of the issue; some dApps don't want to pay for excessive RPC usage.

**Comment by jxom** [comment]
Can you not [set `staleTime`](https://wagmi.sh/react/hooks/useBalance#staletime-optional) instead of doing these workarounds?
